typewritten.zsh

```diff
@@ -65,44 +65,49 @@ tw_get_home_relative_wd() {
   echo "%F{$tw_current_directory_color}%~"
 }
 
-tw_get_git_relative_wd() {
+tw_get_displayed_wd() {
   local tw_home_relative_wd="$1"
   local tw_git_branch="$2"
   local tw_git_home="$3"
```

Comment:
@reobin quick question about these.  Since we're no longer sending `tw_home_relative_wd` as a param, do you feel strongly about no longer sending params to this function in general?  as in changing this block to 

```zsh
  local tw_home_relative_wd="$(tw_get_home_relative_wd)"
  local tw_git_branch="$tw_prompt_data[tw_git_branch]"
  local tw_git_home="$tw_prompt_data[tw_git_home]"
```

Since those aren't used in any other place in `tw_redraw()` except as params for `tw_get_displayed_wd()` it almost makes sense to me that they are just set there but I'm not as versed in zsh as you are so I wanted to make sure this made sense to you as well before I did that.

Reply:
You are right! Good catch

While we're at it, there are some other cleanups we can do in the same section.

The `tw_get_home_relative_wd` function is not very useful. We can merge that into `tw_get_displayed_wd`.

Here's my suggestion for a new `tw_get_displayed_wd`:
```shell
tw_get_displayed_wd() {
  local tw_git_branch=$tw_prompt_data[tw_git_branch]
  local tw_git_home=$tw_prompt_data[tw_git_home]

  local tw_home_relative_wd="%~"
  local tw_git_relative_wd="$tw_git_home%c"

  local tw_displayed_wd="$tw_git_relative_wd"

  # The pure layout defaults to home relative working directory, but allows customization
  if [[ "$TYPEWRITTEN_PROMPT_LAYOUT" = "pure" && "$TYPEWRITTEN_RELATIVE_PATH" = "" ]]; then
    tw_displayed_wd=$tw_home_relative_wd
  fi;

  if [[ "$TYPEWRITTEN_RELATIVE_PATH" = "home" ]]; then
    tw_displayed_wd=$tw_home_relative_wd
  fi;

  if [[ "$TYPEWRITTEN_RELATIVE_PATH" = "adaptive" ]]; then
    if [[ "$tw_git_branch" = "" ]]; then
      tw_displayed_wd=$tw_home_relative_wd
    fi;
  fi;

  echo "%F{$tw_current_directory_color}$tw_displayed_wd"
}
```

Mainly what changes here is that we only make 1 call to the color that the current directory should have, and that's on the return statement (`echo`).

Reply:
Awesome, changes pushed up in c087e32